### PR TITLE
Response resource

### PR DESCRIPTION
**Pull request by RonAzar**
•Added FeedbackResponseResource.kt to handle feedback response submissions.
	•Integrated role and permission checks to ensure authorized access for response submission.

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/ResponseResource.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        val companyId = extractClaimAsLong(requestContext, "companyId")!!
+        val role = extractRole(requestContext)!!
+        val employeeId = extractClaimAsLong(requestContext, "employeeId")!!
+
+        return if (hasPermission(role, Permissions.RESPONSE_TO_FEEDBACK)) {
+            try {
+                val responseId = responseService.submitResponse(newFeedbackResponse, employeeId, companyId)
+                Response.ok("Response submitted successfully, response id: $responseId").build()
+            } catch (e: IllegalArgumentException) {
+                Response.status(Response.Status.BAD_REQUEST).entity(e.message).build()
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/ResponseResource.kt:37]: if you throw a bad request exception in the service, you don't have to do this condition here, the exception will just bubble up and return the relevant response

File: src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/ResponseResource.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            }
+        } else {
+            Response.status(Response.Status.UNAUTHORIZED)
+                .entity("UNAUTHORIZED: You do not have permission to respond to feedbacks").build()
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/ResponseResource.kt:41]: it's cleaner if you do:
if (no permission) return

bla bla rest of the code, not in if block

Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/ResponseResource.kt:35]: Got it!

**Review by vasofi — APPROVED** [review]
same comment here:
remove new lines at the top of the bloks.
Make sure you don't have conflicts

**Pull request merged**